osbrain/tests/test_logging.py

```diff
@@ -24,26 +24,22 @@ def test_logging_level(nsproxy):
     while not len(logger.get_attr('log_history')):
         i += 1
         agent.log_info(i)
-    while not '): %s' % i in logger.get_attr('log_history')[-1]:
-        time.sleep(0.01)
+    assert logger_received(logger, message='\): %s' % i, position=-1)
     before = len(logger.get_attr('log_history'))
     agent.log_info('some information')
     agent.log_warning('some warning')
     agent.log_error('some error')
     agent.log_debug('some debug')
-    after = before
-    while not after - before == 4:
-        after = len(logger.get_attr('log_history'))
-    # Log size
+    # Log values
+    assert logger_received(logger, 'some information', 'log_history_info', -1)
+    assert logger_received(logger, 'some warning', 'log_history_warning', -1)
+    assert logger_received(logger, 'some error', 'log_history_error', -1)
+    assert logger_received(logger, 'some debug', 'log_history_debug', -1)
+    # Log sizes
     assert len(logger.get_attr('log_history_info')) == 1 + before
     assert len(logger.get_attr('log_history_warning')) == 1
     assert len(logger.get_attr('log_history_error')) == 1
     assert len(logger.get_attr('log_history_debug')) == 1
```

Comment:
I would rather have separate assertions so that the errors (if they ever occur) are more clear. In this case it will not save many lines (only the last 3 are expected to have length 1).